### DISCUSSION TITLE
Sporadic Invalid Payload Errors

### DISCUSSION THREAD

**Discussion by clombardi-axiom** [strongswan/strongswan#538] [Q&A]
I am using a Windows 2019 server with a demand dial interface to connect to strongSwan hosted on an Ubuntu 18.04 server and this works "most" of the time. I see sporadic failures with connectivity between the Windows server and the strongSwan VPN server. In the Windows event viewer I always see the same pattern of informational and error events when connectivity is lost. Below is a list of the events and they happen all within the same second. The event captured in the system log is the most telling which states invalid payload.  I know the PSK works because the connection will work again shortly after with no manual intervention. The PSK length is 19 characters if that matters and uses letters, numbers, and special characters. The MTU on the demand dial interface is 1400 and the Ubuntu server is configured for 9001.  Does anyone know what would cause this sporadic connectivity?

**First Event**
CoId={23D1FFE1-8F27-463F-931A-EE6771163A28}: The user SYSTEM has started dialing a VPN connection using a all-user connection profile named 10.45.0.5. The connection settings are: 
Dial-in User = 
VpnStrategy = IKEv2
DataEncryption = Require
PrerequisiteEntry = 
AutoLogon = No
UseRasCredentials = Yes
Authentication Type = PreShareKey 
Ipv4DefaultGateway = Yes
Ipv4AddressAssignment = By Server
Ipv4DNSServerAssignment = By Server
Ipv6DefaultGateway = Yes
Ipv6AddressAssignment = By Server
Ipv6DNSServerAssignment = By Server
IpDnsFlags = 
IpNBTEnabled = No
UseFlags = Private Connection
ConnectOnWinlogon = No
Mobility enabled for IKEv2 = No.

**Second Event**
CoId={23D1FFE1-8F27-463F-931A-EE6771163A28}: The user SYSTEM is trying to establish a link to the Remote Access Server for the connection named 10.45.0.5 using the following device: 
Server address/Phone Number = 10.45.0.5
Device = WAN Miniport (IKEv2)
Port = VPN2-127
MediaType = VPN.

**Third Event**
CoId={23D1FFE1-8F27-463F-931A-EE6771163A28}: The user SYSTEM has successfully established a link to the Remote Access Server using the following device: 
Server address/Phone Number = 10.45.0.5
Device = WAN Miniport (IKEv2)
Port = VPN2-127
MediaType = VPN.

**Fourth Event**
CoId={23D1FFE1-8F27-463F-931A-EE6771163A28}: The link to the Remote Access Server has been established by user SYSTEM.

**Fifth Event (error)**
CoId={23D1FFE1-8F27-463F-931A-EE6771163A28}: The user SYSTEM dialed a connection named 10.45.0.5 which has failed. The error code returned on failure is 13843.

**Error in System Log**
RoutingDomainID- {00000000-0000-0000-0000-000000000000}: A Demand Dial connection to the remote interface 10.45.0.5 on port VPN2-127 was successfully initiated but failed to complete successfully because of the  following error: Invalid payload received

**Error on VPN server in charon log**
06[NET] received packet: from 10.45.3.8[4500] to 10.45.0.5[4500] (1324 bytes)
06[ENC] parsing body of message, first payload is ENCRYPTED
06[ENC] starting parsing a ENCRYPTED payload
06[ENC] parsing ENCRYPTED payload, 1296 bytes left
06[ENC]   parsing rule 0 U_INT_8
06[ENC]   parsing rule 1 U_INT_8
06[ENC]   parsing rule 2 PAYLOAD_LENGTH
06[ENC]   parsing rule 3 CHUNK_DATA
06[ENC] parsing ENCRYPTED payload finished
06[ENC] verifying payload of type ENCRYPTED
06[ENC] ENCRYPTED payload verified, adding to payload list
06[ENC] ENCRYPTED payload found, stop parsing
06[ENC] process payload of type ENCRYPTED
06[ENC] found an encrypted payload
**06[ENC] decrypting encrypted payload payload failed, invalid length**
**06[ENC] could not decrypt payloads**
06[IKE] integrity check failed
06[IKE] CREATE_CHILD_SA request with message ID 0 processing failed
06[MGR] checkin IKE_SA shp1app[18]
06[MGR] checkin of IKE_SA successful
03[NET] waiting for data on sockets
03[NET] received packet: from 10.45.3.8[4500] to 10.45.0.5[4500]
03[ENC] parsing header of message
03[ENC] parsing HEADER payload, 1324 bytes left
03[ENC]   parsing rule 0 IKE_SPI
03[ENC]   parsing rule 1 IKE_SPI
03[ENC]   parsing rule 2 U_INT_8
03[ENC]   parsing rule 3 U_INT_4
03[ENC]   parsing rule 4 U_INT_4
03[ENC]   parsing rule 5 U_INT_8
03[ENC]   parsing rule 6 RESERVED_BIT
03[ENC]   parsing rule 7 RESERVED_BIT
03[ENC]   parsing rule 8 FLAG
03[ENC]   parsing rule 9 FLAG
03[ENC]   parsing rule 10 FLAG
03[ENC]   parsing rule 11 FLAG
03[ENC]   parsing rule 12 FLAG
03[ENC]   parsing rule 13 FLAG
03[ENC]   parsing rule 14 U_INT_32
03[ENC]   parsing rule 15 HEADER_LENGTH
03[ENC] parsing HEADER payload finished
03[ENC] parsed a CREATE_CHILD_SA request header
09[MGR] checkout IKEv2 SA by message with SPIs e8ecd6801d61ac5b_i f7d2318e8a673c0d_r
09[MGR] IKE_SA shp1app[18] successfully checked out

**ipsec.conf**

config setup
  strictcrlpolicy=no
  uniqueids=no

conn %default
  type=tunnel
  ikelifetime=1440m                 # 24 hours
  lifetime=480m                     # 8 hours
  keylife=60m
  rekeymargin=3m
  keyingtries=1
  keyexchange=ikev1
  authby=secret
  dpddelay=30
  dpdtimeout=150
  dpdaction=clear

include /etc/ipsec.d/*.conf

**WindowsServer2019.conf**
conn windowsserver2019
  auto=route
  fragmentation=yes
  left=10.45.0.5
  leftid=10.45.0.5
  leftsubnet=10.10.10.0/24,192.168.0.0/24,192.168.253.0/24,192.168.251.0/24,172.23.109.0/24,172.31.9.0/24, 100.111.9.0/24
  leftauth=psk
  right=10.45.4.11
  rightsubnet=10.45.4.11/32
  rightid=10.45.4.11
  rightauth=psk
  keyexchange=ikev2
  ike=aes256-sha1-modp1024         # P1: modp2048 = DH group 14
  esp=aes256-sha1                   # P2

**Comment by Thermi** [comment]
Well, looks like Windows Server is not putting together correct IKE packets. If you want, you can provide a packet capture (in pcap or pcap-ng format, please) of the IKE exchange, then we can see it directly.

**Reply by clombardi-axiom** [reply]
Thank you. I am performing a capture now, but will need to wait for the issue to show itself again since I cannot replicate it on command.

**Reply by clombardi-axiom** [reply]
Well this is weird. I started a trace with no filters last night and kept my RDP session open. This morning I reviewed the event viewer logs and see the usual sequence of informational events and then the error event starting at 22:28 and it errored consistently to 03:53 this morning. What is weird is that there is zero network traffic in that gap (see below). Network traffic started again at the same time that the demand dial interface was able to successfully connect. I was expecting to see some outbound requests to the VPN gateway, but I don't see a thing. 

![capture](https://user-images.githubusercontent.com/88325520/128191607-1cd4d870-d5ae-4583-89e0-59a73e27904a.JPG)

**Reply by Thermi** [reply]
Well, you need to capture on the physical interface, not on the WAN miniport.

**Reply by clombardi-axiom** [reply]
[ipsectracetwo.zip](https://github.com/strongswan/strongswan/files/6935245/ipsectracetwo.zip)
See attached. According to event viewer the errors started at 15:21:32, 15:22:24, 15:24:04, 15:24:11, and in random intervals until the last error at 16:38:26.

**Reply by Thermi** [reply]
Do you have logs accompanying that?

**Reply by clombardi-axiom** [reply]
[aug5_5.zip](https://github.com/strongswan/strongswan/files/6940950/aug5_5.zip)
[charonfortracethree.txt](https://github.com/strongswan/strongswan/files/6940951/charonfortracethree.txt)

Attached are new logs and the associated charon log for that time period. It turns out the server that is having the problem (10.45.4.11) has a three minute and some additional seconds time delay. It is not sync'd with an NTP server. I time shifted the output of the trace by +3 seconds to try and make it easier to read. In the charon logs the traffic that coincided with the error started at 05:59:16
